phpBB/phpbb/auth/provider/oauth/oauth.php

```diff
@@ -539,109 +471,219 @@ public function link_account(array $link_data)
 				return $this->link_account_auth_link($link_data, $service_name);
 			case 'login_link':
 				return $this->link_account_login_link($link_data, $service_name);
+			default:
+				return 'LOGIN_LINK_MISSING_DATA';
 		}
 	}
 
 	/**
-	* Performs the account linking for login_link
-	*
-	* @param	array	$link_data		The same variable given to {@see \phpbb\auth\provider\provider_interface::link_account}
-	* @param	string	$service_name	The name of the service being used in
-	*									linking.
-	* @return	string|null	Returns a language constant (string) if an error is
-	*						encountered, or null on success.
-	*/
+	 * {@inheritdoc}
+	 */
+	public function logout($data, $new_session)
+	{
+		// Clear all tokens belonging to the user
+		$storage = new token_storage($this->db, $this->user, $this->oauth_token_table, $this->oauth_state_table);
```

Comment:
Can't you use the container for this?

Reply:
I think using the container for this does not make too much sense as it can't be shared between users and has to be recreated multiple times depending on session changes. Therefore I think this is ok for now.